pkg/transaction/eventgenerator.go

```diff
@@ -191,6 +195,57 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
 	return events, nil
 }
 
+// getConsumerInfo - get the consumer information to add to transaction event
+func (e *Generator) getConsumerInfo(summaryEvent LogEvent) *ConsumerDetails {
+	cacheManager := agent.GetCacheManager()
+	appName := ""
+
+	if summaryEvent.TransactionSummary.Application != nil {
+		appName = summaryEvent.TransactionSummary.Application.Name
+	}
+
+	// get proxy information
+	apiID := summaryEvent.TransactionSummary.Proxy.ID
+	stage := summaryEvent.TransactionSummary.Proxy.Stage
```

Comment:
Should we check to make sure all these objects are set? Possible NPE?

Reply:
i've seen the stage come back "" (blank)...  i can check to make sure.. .

Reply:
stage can be blank. I guess we can add a nil check for summaryEvent.TransactionSummary and summaryEvent.TransactionSummary.Proxy